$OpenBSD: patch-libbacktrace_backtrace_h,v 1.2 2012/09/27 15:45:42 jasper Exp $

- BT_MAX_DEPTH and struct bt_frame should not be exposed in a public header
  since they're internal to libbacktrace.

--- libbacktrace/backtrace.h.orig	Fri Sep 10 00:23:45 2010
+++ libbacktrace/backtrace.h	Wed Sep 26 12:54:25 2012
@@ -16,18 +16,26 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 
+#ifndef _EXECINFO_H_
+#define _EXECINFO_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #ifndef __GNUC__
 #error "this library must be compiled with gcc"
 #endif
 
-#define BT_MAX_DEPTH            (128)
+#include <dlfcn.h>
 
-struct bt_frame {
-	Dl_info			bt_dlinfo;
-	unsigned int		line;
-};
-
 /* compatability functions with libexecinfo and glibc */
 int			backtrace(void **, int);
 char			**backtrace_symbols(void *const *, int);
 void			backtrace_symbols_fd(void *const *, int, int);
+
+#ifdef __cplusplus
+}
+#endif
+#endif
+
